client/src/components/ProfilePage/index.js

```diff
@@ -65,11 +70,107 @@ const AccordionDetails = withStyles((theme) => ({
   const [expanded, setExpanded] = React.useState("panel1");
   const { user } = React.useContext(UserContext);
-  // console.log(user);
   const classes = useStyles();
+  
+  const [listings, setListings] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+
+  const fetchItems = async () => {
+    setLoading(true);
+    const itms = await getUserItems(user._id);
+    if (itms.success) {
+      setListings(itms.listings);
+      setError(false);
+    } else setError(true);
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    fetchItems();
+  }, []);
+
+  const StyledText = styled.h1`
+  background-image: linear-gradient(#2196F3,#21CBF3);
+  font-size: 2.5rem;
+  -webkit-background-clip: text;
+  -webkit-text-fill-color: transparent;
+  margin: 0;
+  margin-left: 1rem;
+`;
+
+  const goToItem = (id) => props.history.push(`/item/${id}`);
+  
   const handleChange = (panel) => (event, newExpanded) => {
     setExpanded(newExpanded ? panel : false);
   };
+
+  //if user has no listings
+  if(listings.length == 0)
+  {
+    return (
+      <div>
+        <Accordion
+          square
+          expanded={expanded === "panel1"}
+          onChange={handleChange("panel1")}
+        >
+          <AccordionSummary aria-controls="panel1d-content" id="panel1d-header">
+            <Typography>Username</Typography>
+          </AccordionSummary>
+          <AccordionDetails>
+          <Typography>{user.username}</Typography>
+          </AccordionDetails>
+        </Accordion>
+        <Accordion
+          square
+          expanded={expanded === "panel2"}
+          onChange={handleChange("panel2")}
+        >
+          <AccordionSummary aria-controls="panel2d-content" id="panel2d-header">
+            <Typography>Email</Typography>
+          </AccordionSummary>
+          <AccordionDetails>
+            <Typography>{user.email}</Typography>
+          </AccordionDetails>
+        </Accordion>
+        <Accordion
+          square
+          expanded={expanded === "panel3"}
+          onChange={handleChange("panel3")}
+        >
+          <AccordionSummary aria-controls="panel3d-content" id="panel3d-header">
+            <Typography>Password</Typography>
+          </AccordionSummary>
+          <AccordionDetails>
+            <Typography>{user.password}</Typography>
+          </AccordionDetails>
+        </Accordion>
+        
+        <h1 className="Title">
+          <StyledText>You have no items up for sale</StyledText>
+        </h1>
+  
+        <Button
+          className={classes.button}
+          variant="contained"
+          color="secondary"
+          onClick={() => {user.username = ""
+                          user.email = ""
+                          user.password = ""
+                          user._id = ""
+                          user.authId = ""
+                          user.loggedIn = false
+                          props.history.push("/login")}}
```

Comment:
dylan mentioned this I believe but making&using a setUser by grabbing the userContext would make this code look more clean

Reply:
I updated it so now uses setUser instead.